src/layout/GridList/index.tsx

```diff
@@ -1,42 +1,50 @@
-import React, { PureComponent } from 'react';
+import React from 'react';
 import styled from 'styled-components';
 
 import { media } from '../../core/BreakPoints';
 import { HTMLDivProps } from '../../interfaces/props';
-import { createUniqIDGenerator } from '../../utils/createUniqIDGenerator';
+import { keyExtractor, memoComponent } from '../../utils';
 
 const sizeToPercent = (column?: number) => 100 / (column || 1);
-const generateID = createUniqIDGenerator('grid-list-');
 
-type Column = 1 | 2 | 3 | 4 | 6 | 12;
-interface Props {
-  items: any[] | ReadonlyArray<any>;
-  renderItem: any;
-  smColumn: Column;
-  lgColumn?: Column;
+export type Column = 1 | 2 | 3 | 4 | 6 | 12;
+export interface GridProps<T> {
   className?: string;
   divAttributes?: HTMLDivProps;
+  items: T[] | ReadonlyArray<T>;
+  keyExtractor?: (item: T, index: number) => string;
+  lgColumn?: Column;
+  renderItem: any;
+  smColumn: Column;
 }
 
-export class GridList extends PureComponent<Props> {
-  public render() {
-    const { items, renderItem, smColumn, lgColumn, className, divAttributes } = this.props;
-
+export const GridList = memoComponent(
+  <T extends { [key in string]: any }>({
```

Comment:
```suggestion
  <T>({
```

items에 object[]가 아니라 number[]나 string[] 같은 것도 올 수 있으니...!

Reply:
기존에 사용하던 GridList 코드를 일일이 바꾸지 않게ㅐ 하고 싶다면 `<T = any>` 를 넣어주면 좋을 듯